sdcm/nemesis.py

```diff
@@ -1115,6 +1115,41 @@ def disrupt_network_start_stop_interface(self):  # pylint: disable=invalid-name
         finally:
             self.target_node.remoter.run("sudo /sbin/ifup eth1")
 
+    def disrupt_streaming_error(self):
+        """
+        Stop decommission in middle to trigger some streaming fails, then rebuild the data on the node.
+        If the node is decommission unexpectedly, need to re-add a new node to cluster.
+        """
+        self._set_current_disruption('StreamingError')
+        target_node_ip = self.target_node.ip_address
+
+        def _decommission():
+            try:
+                self.target_node.run_nodetool("decommission")
+            except Exception:  # pylint: disable=broad-except
+                self.log.debug('Decommission is stopped')
+        log_start = self.target_node.mark_log()
+        decommission_thread = threading.Thread(target=_decommission)
+        decommission_thread.start()
+        time.sleep(10)
+        decommission_thread.join(0)
+        self.target_node.remoter.run('pkill -f decommission')
```

Comment:
Is this our best way to intrrupt the decommisison?
Maybe we need to test several ways of intrupttions, like (hard) rebooting a source node, (hard) rebooting a target node, networking issue, etc.
Having said that, we need to make sure we aren't entering into a "parallel nemesis mode" and cluster is still able to serve QUORUM queries.

Reply:
> Is this our best way to interrupt the decommission?
> Maybe we need to test several ways of interruptions, like (hard) rebooting a source node, (hard) rebooting a target node, networking issue, etc.

hard reboot of i3 will cause repair of the reboot node itself. the decommission will fail for stream error. Maybe we need to wait the repair of reboot finishes first, then start rebuild of the decommission node.

> Having said that, we need to make sure we aren't entering into a "parallel nemesis mode" and cluster is still able to serve QUORUM queries.

Reply:
No it won't, restart will cause this, hard reboot will just reboot the server.

Reply:
You are right.